### ISSUE TITLE
[1.10.2] IC Gate state gets screwed up when saving and loading a level

### ISSUE THREAD

**Issue by EnderShadow** [MrTJP/ProjectRed#1245]
I created an 16 bit programmable binary decoder which acts weird upon placement and world unload/load. For some reason seemingly random wires get activated in the circuit even though there should be no reason for them to be active. The only way I've found to fix it is to enable and disable all inputs, but the problem re-occurs after the world is unloaded and reloaded. From what I can tell the issue is caused by gates that invert their input. The issue occurs with schematics too. Here's an image of the schematic after a world reload. [https://i.imgur.com/ZnWCaiw.png](https://i.imgur.com/ZnWCaiw.png?1)

The gates on the right are NOT gates, the gates on the lower diagonal are XOR gates, and the rest of the gates are AND gates. The bottom and right are inputs while the top is an output

I have tested this with NOT, NAND, and NOR gates

**Comment by evg-zhabotinsky** [comment]
I confirm this. I've built a chip with 16 8-bit dual read registers, and upon placement or world load it seems to get stuck in places where it should not be able to. It reliably starts functioning properly only after I wipe it twice, first with all ones then with all zeroes, and read it whole on both channels after both passes. I've even built a chip that performs that procedure, but it has to be triggered twice before it starts working properly.

Also worth a note that the effect seems to be quite random, while also depending on circuit state before it was unloaded. After the wipe procedure described above and reloading game world, I haven't once seen any outputs stuck in active state on that register bank chip (though it still does not really work before another wipe), while having stored any data in it beforehand pretty much guarantees that some outputs will get stuck high.

**Comment by opendata26** [comment]
Yep, I also get this with my adder

**Comment by MatCat** [comment]
Also experience this issue, I accidentally created another issue thread #1448 because I did not see this thread when I looked to see if anyone else had reported it.

**Comment by MrTJP** [comment]
This bug still needs tracking down. Am queuing it as high priority. Most likely not the same reason it was happening in 1.10.2, since the entire system was rewritten recently.
